core/shared/platform/nuttx/nuttx_platform.c

```diff
@@ -182,6 +187,11 @@ os_get_dbus_mirror(void *ibus)
     if (in_ibus_ext(ibus)) {
         return (void *)((uint8 *)ibus - MEM_DUAL_BUS_OFFSET);
     }
+    else if (INTERNAL_SRAM_1_IBUS_ADDRESS_LOW <= (uintptr_t)ibus
+             && (uintptr_t)ibus < INTERNAL_SRAM_1_IBUS_ADDRESS_HIGH) {
+        return (void *)((uintptr_t)ibus - INTERNAL_SRAM_1_IBUS_ADDRESS_LOW
```

Comment:
yes.
for some configurations, os_mmap can return memory from the internal ram 1.

Reply:
Will the special handling be implemented in this PR?

Reply:
sorry, which special handling are you talking about?

Reply:
Can we move out these arch specific impls from common place?

Reply:
> sorry, which special handling are you talking about?

I mean if the os_mmap returns memory from the internal ram1, does that also need convert it into its mirrored addr once the feature enabled?

Reply:
up_textheap_memalign already returns i-bus address.

i have no idea about the WASM_MEM_DUAL_BUS_MIRROR block in os_mmap.
to me, it seems like a dead code because CONFIG_ARCH_USE_TEXT_HEAP is always set for
esp32s3.
i guess it's different for the private fork of nuttx?

Reply:
> Can we move out these arch specific impls from common place?

i guess it can be eventually moved to nuttx.

Reply:
> to me, it seems like a dead code because CONFIG_ARCH_USE_TEXT_HEAP is always set for

OK, agree to refactor this into nuttx.